src/Moryx.ControlSystem/Processes/ProcessHolderExtensions.cs

```diff
@@ -11,121 +11,117 @@ namespace Moryx.ControlSystem.Processes
 
         /// <summary>
         /// Get the position by id of the running activity
         /// </summary>
         public static TPosition GetPositionByActivityId<TPosition>(this IProcessHolderGroup<TPosition> group, long activityId)
-            where TPosition : IProcessHolderPosition
-        {
-            return GetPosition(group.Positions, t => (t.Session as ActivityStart)?.Activity.Id == activityId);
-        }
+            where TPosition : IProcessHolderPosition => GetPosition(group.Positions, t => (t.Session as ActivityStart)?.Activity.Id == activityId);
 
         /// <summary>
-        /// Checks if the group holds a process with a finished activity having the matching result
-        /// </summary>
-        public static bool HasFinishedActivity<TPosition>(this IProcessHolderGroup<TPosition> group, long activityId, long activityResult)
-            where TPosition : IProcessHolderPosition
-        {
-            return group.Positions.Any(position => position.HasFinishedActivity(activityId, activityResult));
-        }
-        /// <summary>
-        /// Checks if the position holds a process with a finished activity having the matching result.
+        /// Get the position by id of the running activity
         /// </summary>
-        public static bool HasFinishedActivity(this IProcessHolderPosition holderPosition, long activityId, long activityResult)
-        {
-            return holderPosition.Process?.GetActivities(activity => activity.Id == activityId && activity.Result?.Numeric == activityResult).Any() == true;
-        }
+        public static IProcessHolderPosition GetPositionByActivityId(this IProcessHolderGroup group, long activityId)
+            => GetPosition(group.Positions, t => (t.Session as ActivityStart)?.Activity.Id == activityId);
 
         /// <summary>
         /// Get the position by id of the running activity
         /// </summary>
         public static TPosition GetPositionByActivityId<TPosition>(this IEnumerable<TPosition> positions, long activityId)
-            where TPosition : IProcessHolderPosition
+            where TPosition : IProcessHolderPosition => GetPosition(positions, t => (t.Session as ActivityStart)?.Activity.Id == activityId);
+
+        /// <summary>
+        /// Tries to get an empty <see cref="IProcessHolderPosition"/> from the <paramref name="group"/>
+        /// </summary>
+        /// <param name="group">The group to search</param>
+        /// <param name="position">When this method returns, contains the first empty 
+        /// <see cref="IProcessHolderPosition"/>, if any is found; otherwise <c>null</c></param>
+        /// <returns>
+        /// <c>true</c> if an empty <see cref="IProcessHolderPosition"/> was found; otherwise, <c>false</c>.
+        /// </returns>
+        public static bool TryGetEmptyPosition(this IProcessHolderGroup group, out IProcessHolderPosition position)
```

Comment:
Could you give me an example for that? Would you want to get the position as a parameter in that action or execute the action on the position or is it just an action if an empty position is available? 🤔

Reply:
A combination of 
> Would you want to get the position as a parameter in that action
and 
> just an action if an empty position is available

Like
```cs
public static void CheckEmptyPosition(this IProcessHolderGroup group, Action<IProcessHolderPosition> action)
{
    ...
    if(postion.IsEmpty())
        action(position);
}
```

Reply:
What would you say about a `TryOnEmptyPosition` method, would that cover your use case? 🤔
```
public static void TryOnEmptyPosition(this IProcessHolderGroup group, Action<IProcessHolderPosition> action, out IProcessHolderPosition position)
{
    position = group.GetEmptyPosition()
    if(position is null)
        return false;
    action(position);
    return true;
}
```

Reply:
Would your trough an exception if there is no empty position then? 🤔

Reply:
No, it should just offer the 'positive' path and remove all the 'exception handling' as you would probably use it like this:

```
group.OnEmptyPosition(position => DoSomethingWithIt);
```

Question is: Do you have the need to handle the case when there is no empty position?

Reply:
I would argue that you have, if there is no empty position you most likely have a problem in your Resource. For example, let's say you want to create a RTW on an empty position for a given process id you recieved from a driver. If there is no empty position, you most likely have a serious issue somewhere that you should be notified of or you expect that behaviour but would still need to respond to the other case as you have to do something with the process you just recieved 😅

Reply:
Ok, I understand. Then, I would still not mix all those concepts but maybe provide another Action parameter, that gets invoked in case of an error, or you could think about returning a [FunctionResult](https://github.com/PHOENIXCONTACT/MORYX-Framework/blob/main/src/Moryx/Tools/FunctionResult.cs)

Reply:
Is it alright if I leave it as is for now and we add you suggestion when we have a concrete example? I must admit I don't have a feeling whether your suggestion would actually simplify the code/decision flow and would like to see it first 🙈
The current proposal is what I would use similar to .net methods like the [TryDeque](https://learn.microsoft.com/de-de/dotnet/api/system.collections.concurrent.concurrentqueue-1.trydequeue?view=net-8.0)